indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts

```diff
@@ -1,63 +1,230 @@
+    }
+    logger.info({
+      at: 'aggregate-trading-rewards#runTask',
+      message: 'Generated interval to aggregate trading rewards',
+      start: interval.start.toISO(),
+      end: interval.end.toISO(),
+    });
 
-function sortTradingRewardData(
-  _tradingRewardData: TradingRewardFromDatabase[],
-): SortedTradingRewardData {
-  // TODO: Implement
-  return {};
-}
+    const tradingRewardData:
+    TradingRewardFromDatabase[] = await this.getTradingRewardDataToProcess(interval);
+    const sortedTradingRewardData: SortedTradingRewardData = this.sortTradingRewardData(
+      tradingRewardData,
+    );
+    await this.updateTradingRewardsAggregation(sortedTradingRewardData);
+    await AggregateTradingRewardsProcessedCache.setProcessedTime(
+      this.period,
+      interval.end.toISO(),
+      redisClient,
+    );
+  }
+
+  /**
+   * Gets the interval of time to aggregate trading rewards for.
+   * If there are no blocks in the database, then do not process any data.
+   * If There is no processedTime in the cache, then delete the latest month of data,
+   * and reprocess that data or start from block 1.
+   * If the processedTime is not null, and blocks exist in the database, then process up to the
+   * next hour of data.
+   */
+  async getTradingRewardDataToProcessInterval(): Promise<Interval | undefined> {
+    let latestBlock: BlockFromDatabase;
+    try {
+      latestBlock = await BlockTable.getLatest();
+    } catch (e) {
+      logger.info({
+        at: 'aggregate-trading-rewards#getTradingRewardDataToProcessInterval',
+        message: 'Unable to aggregate trading rewards because there are no blocks in the database.',
+      });
+      return;
+    }
+
+    const processedTime:
+    IsoString | null = await AggregateTradingRewardsProcessedCache.getProcessedTime(
+      this.period,
+      redisClient,
+    );
+    if (processedTime === null) {
+      await this.deleteIncompleteAggregatedTradingReward();
+      logger.info({
+        at: 'aggregate-trading-rewards#getTradingRewardDataToProcessInterval',
+        message: 'Resetting AggregateTradingRewardsProcessedCache',
+      });
+      const nextStartTime: DateTime = await this.getNextIntervalStart();
+      await AggregateTradingRewardsProcessedCache.setProcessedTime(
+        this.period,
+        nextStartTime.toISO(),
+        redisClient,
+      );
+
+      return this.generateInterval(nextStartTime, latestBlock);
+    }
+
+    const startTime: DateTime = DateTime.fromISO(processedTime, UTC_OPTIONS);
+    return this.generateInterval(startTime, latestBlock);
+  }
+
+  /**
+   * Deletes the latest this.period of aggregated trading rewards if it is incomplete. This is
+   * called when the processedTime is null, and we need to reprocess the latest month of data.
+   * We delete the latest this.period of aggregated trading rewards data because we don't know if
+   * the data is complete.
+   */
+  private async deleteIncompleteAggregatedTradingReward(): Promise<void> {
```

Comment:
I think the nuance here is that when writing to the `TradingRewardAggregationTable`, you only set `endedAt` if it is a full interval & not mid-interval. Can you call this out in the comments?
Also, consider renaming this function to `potentiallyDeleteIncompleteAggregatedTradingReward`? since it's not guaranteed you are doing a deletion